dspace-api/src/main/java/org/dspace/identifier/FilteredIdentifierProvider.java

```diff
@@ -0,0 +1,93 @@
+ * handle "skip filter" booleans, so that any configured filters can be skipped and DOI registration forced.
+ *
+ * @author Kim Shepherd
+ * @version $Revision$
+ */
+public abstract class FilteredIdentifierProvider extends IdentifierProvider {
+
+    protected Filter filterService;
+
+    /**
+     * Setter for spring to set the filter service from the property in configuration XML
+     * @param filterService - an object implementing the org.dspace.content.logic.Filter interface
+     */
+    @Autowired
+    public void setFilterService(Filter filterService) {
+        this.filterService = filterService;
+    }
+
+    /**
+     * Register a new identifier for a given DSpaceObject
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject to use for identifier registration
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before performing registration
+     * @return identifier
+     * @throws IdentifierException
+     */
+    public abstract String register(Context context, DSpaceObject dso, Boolean skipFilter)
+        throws IdentifierException;
+
+    /**
+     * Register a specified identifier for a given DSpaceObject
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject identified by the new identifier
+     * @param identifier - String containing the identifier to register
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before performing registration
+     * @throws IdentifierException
+     */
+    public abstract void register(Context context, DSpaceObject dso, String identifier, Boolean skipFilter)
+        throws IdentifierException;
+
+    /**
+     * Reserve a specified identifier for a given DSpaceObject (eg. reserving a DOI online with a registration agency)
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject identified by this identifier
+     * @param identifier - String containing the identifier to reserve
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before performing reservation
+     * @throws IdentifierException
+     * @throws IllegalArgumentException
+     * @throws SQLException
+     */
+    public abstract void reserve(Context context, DSpaceObject dso, String identifier, Boolean skipFilter)
+        throws IdentifierException, IllegalArgumentException, SQLException;
+
+    /**
+     * Mint a new identifier in DSpace - this is usually the first step of registration
+     * @param context    - DSpace context
+     * @param dso        - DSpaceObject identified by the new identifier
+     * @param skipFilter - boolean indicating whether to skip any filtering of items before minting
+     * @return a String containing the new identifier
+     * @throws IdentifierException
+     */
+    public abstract String mint(Context context, DSpaceObject dso, Boolean skipFilter) throws IdentifierException;
```

Comment:
better to use a primitive boolean

Reply:
see my other comment near the canMint section - a null or missing filter should not default to false, so I didn't want to use a primitive here

Reply:
ok but better to make this explicit in the javadoc

Reply:
I had a further think about this and realised primitive is fine - this is the skipFilter boolean anyway, not result (which was already refactored to primitive!) So i've refactored to primitive boolean and tested, we can resolve this.